src/b-splines/linear.jl

```diff
@@ -1,4 +1,14 @@
-struct Linear <: Degree{1} end  # boundary conditions not supported
+struct Linear{BC<:Union{Throw{OnGrid},Periodic{OnCell}}} <: DegreeBC{1}
+    bc::BC
+end
+
+(deg::Linear)(gt::GridType) = Linear(deg.bc(gt))
```

Comment:
Do we actually need this method?
```julia
julia> L = Linear()
Linear()

julia> L(OnGrid())
Linear()

julia> L(OnCell())
ERROR: MethodError: no method matching Linear(::Throw{OnCell})
...
```

Reply:
I agree, we do not need this method. (It was a leftover from copying from cubic.)